gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/resolutionstrategy/ComponentSelectionWithCurrent.kt

```diff
@@ -4,7 +4,7 @@ import org.gradle.api.artifacts.ComponentSelection
 
 class ComponentSelectionWithCurrent(
   private val delegate: ComponentSelection,
-  val currentVersion: String? = null,
+  val currentVersion: String,
```

Comment:
This looks like a safe change from the internal use of the class, but I'm not sure if there could be some external consumers relying on this being nullable for some reason.

Reply:
Some dependency types lack versions, like files. I’d be more concerned about if that was affected than external usages of our internal classes.

Reply:
Yes, those seem to be handled with a default "none" value rather than with null (https://github.com/ben-manes/gradle-versions-plugin/blob/master/gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Coordinate.kt#L28). 

The only place where `ComponentSelectionWithCurrent` instances are created is https://github.com/ben-manes/gradle-versions-plugin/blob/master/gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/resolutionstrategy/ComponentSelectionRulesWithCurrent.kt#L84, and at that point the `current` variable can't be null (as there is `?: return null` check right before), and `Coordinate.version` is not-null (https://github.com/ben-manes/gradle-versions-plugin/blob/master/gradle-versions-plugin/src/main/kotlin/com/github/benmanes/gradle/versions/updates/Coordinate.kt#L20).

Let me know if that makes sense to you.